### PR TITLE
Auto-completion for tags field

### LINKED ISSUE #29: Add autocompletion for tags
To make it easier to re-use existing tags, it would be great to some form of autocompletion.

### PR DESCRIPTION
**Pull request by skddc**
* Cache existing tags in localStorage when all bookmarks are loaded
* Read from localStorage and use with new autocomplete UI when adding or editing a bookmark
* Delete cache key when disconnecting storage

closes #29

### CODE REVIEW

**Comment by skddc** [comment]
I think this is done for the basic functionality. Would appreciate a quick review though.

File: app/controllers/application.js [hunk 1]
```diff
@@ -11,7 +11,8 @@ export default Ember.Controller.extend({
   },
 
   handleStorageDisconnect: function() {
-    console.log('rs disconnected, transition to welcome');
+    localStorage.removeItem('webmarks:tags');
```
Comment (galfert) [inline app/controllers/application.js]: You might want to move this into the `storage` service as well, similar to all other `localStorage` operations.

**Comment by skddc** [comment]
> When removing a bookmark with tags that are only used for that bookmark, it might be a good idea to delete those tags from the cache.

They'll be overwritten entirely next time you load the index, hence the todo method also for adding tags.

> Also, when adding tags via autocompletion the last trailing comma results in an empty tag being added.

The auto-completion is only putting things in the input field, so apparently there's a bug with trailing commas in general.

File: app/controllers/application.js (hunk 1 above)
Comment (skddc) [inline app/controllers/application.js]: Good idea. Done.

**Comment by skddc** [comment]
I moved the tag list deletion to the service and fixed the bug with trailing slashes. Thanks for your review!

**Pull request merged**